models/ArticleManager.php

```diff
@@ -0,0 +1,377 @@
+                    admins.name
+            FROM articles
+            JOIN admins
+            ON articles.writer = admins.email
+            WHERE articles.id = '{$_GET['id']}'
+        ");
+     $post = [];
+        while($row = $req->fetch()){
+            
+            $post   = new Article();
+            $post   ->setId($row['id']);
+            $post   ->setTitle($row['title']);
+            $post   ->setChapo($row['chapo']);
+            $post   ->setContent($row['content']);
+            $post   ->setDate($row['date']);
+            $post   ->setImage($row['image']);
+                        
+            $posts[] = $post;
+        }
+        return $post;
+    }
+
+    // Fonction permettant d'éditer un article 
+    function edit($title,$chapo,$content,$posted,$id){
+        $tableau = [
+            'title'     => $title,
+            'chapo'     => $chapo,
+            'content'   => $content,
+            'posted'    => $posted,
+            'id'        => $id
+        ];
+
+        $sql = "UPDATE articles SET title=:title, chapo=:chapo, content=:content, date=NOW(), posted=:posted WHERE id=:id";
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+    }
+
+    // Fonction permettant d'insérer un nouvel article dans la base de données
+    function post($title, $chapo, $content,$posted){
+        $tableau = [
+            'title'     =>  $title,
+            'chapo'     =>  $chapo,
+            'content'   =>  $content,
+            'writer'    =>  $_SESSION['admin'],
+            'posted'    =>  $posted
+        ];
+
+        $sql = "
+        INSERT INTO articles(title, content, chapo, writer, date, posted, image)
+        VALUES(:title,:content,:chapo,:writer,NOW(),:posted, '')
+        ";
+
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+    }
+
+    // Fonction permettant d'insérer une nouvelle image dans la base de données
+    function post_img($tmp_name, $extension){
+        $id = $this->db->lastInsertId(); //On doit mettre la fonction lastInsertId seulement après une fonction INSERT.
+        
+        $tableau = [
+            'id'    =>  $id,
+            'image' =>  $id.$extension  
+        ];
+
+        $sql = "UPDATE articles SET image = :image WHERE id = :id";
+        
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+        move_uploaded_file($tmp_name, "public/img/posts/".$id.$extension);
+    }
+
+        // Fonction permettant de mettre à jour l'image d'un article
+    function update_img($tmp_name, $extension){
+        $id = filter_input(INPUT_GET, 'id', FILTER_SANITIZE_NUMBER_INT);
+         
+        $tableau = [
+            'id'    =>  $id,
+            'image' =>  $id.$extension  
+        ];
+
+        $sql = "UPDATE articles SET image = :image WHERE id = :id";
+        
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+        move_uploaded_file($tmp_name, "public/img/posts/".$id.$extension);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function move_uploaded_file() is discouraged](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)